src/apps/replication/lib/replication_failure_detector.h

```diff
@@ -57,11 +57,11 @@ class replication_failure_detector  : public dsn::fd::failure_detector
     virtual void on_worker_disconnected( const std::vector< ::dsn::rpc_address>& nodes ) { dassert (false, ""); }
     virtual void on_worker_connected( ::dsn::rpc_address node )  { dassert (false, ""); }
 
-    ::dsn::rpc_address current_server_contact() const { zauto_lock l(_meta_lock); return dsn_group_get_leader(_meta_servers.group_handle()); }
+    ::dsn::rpc_address current_server_contact() const { zauto_lock l(failure_detector::_lock); return dsn_group_get_leader(_meta_servers.group_handle()); }
     ::dsn::rpc_address get_servers() const  { return _meta_servers; }
```

Comment:
there is indeed an issue here - the lock is not needed, but instead, we should make group address thread safe - i'm working on it.